### PR TITLE
Implement Search and Filter functionality on Job Postings Page

### PR DESCRIPTION
**Pull request by nikkikapadia**
Added APIs on the backend for retrieving jobs, companies and user data from db. Changed frontend code to get data from the backend instead of mock data file. Fixed up the frontend due to issues with original logic given new data source.

NOTE: we're only calling user 1 at the moment but once proper routing from login is implemented it will be able to call for the user that is signed in.

### CODE REVIEW

File: launchpad_backend/launchpad_server/routes/index.py [hunk 1]
```diff
@@ -6,7 +6,8 @@
 from flask_pymongo import PyMongo
 import pymongo
 import sys
-sys.path.append('.\\launchpad_server\\routes')
+import re
+sys.path.append('./launchpad_server/routes')
```
Comment (nikkikapadia) [inline launchpad_backend/launchpad_server/routes/index.py:10]: can someone check if this works on a windows environment? The \\ wasn't working on my laptop.

**Review by al-yanna — APPROVED** [review]
LGTM 🚀 

**note for future reference:**  `/jobs` endpoint will return nothing, if you want the full list of jobs (no filters) do `/jobs?type=null&duration=null&location=null`

**Review by BrianBNguyen — COMMENTED** [review]
Functionality works on my end. I'm Getting the following console warning below though. 

image

**Review by BrianBNguyen — APPROVED** [review]
Warning are minor. I don't think it'll affect anything. Everything else looks good! Thanks :)

File: launchpad_backend/launchpad_server/routes/index.py (hunk 1 above)
Comment (BrianBNguyen) [reply launchpad_backend/launchpad_server/routes/index.py:10]: LGTM

**Pull request merged**